src/app/modules/submission/encrypt-submission/encrypt-submission.controller.ts

```diff
@@ -349,10 +349,7 @@ const submitEncryptModeForm: ControllerHandler<
   }
 
   // Handle submissions for payments forms
-  if (
-    form.payments_field?.enabled &&
-    form.payments_channel.channel === PaymentChannel.Stripe
-  ) {
+  if (form.payments_channel.channel === PaymentChannel.Stripe) {
```

Comment:
This concerns me somewhat. Since we no longer have a singular `enabled` boolean, the payment channel can be connected, but amount and or description not set, and thus payment is actually not enabled on the form for the incoming submission, correct?

Basically I note a distinction between making the admins able to update all the payment fields (or not) in the form definition, independently and without a `enabled` toggle gating the fields; but to display the form to respondent, and validate incoming submissions, we need to be extra careful that payment is indeed enabled.

Reply:
Yep, that's why there's the check everywhere for all the 3 requirements (connection, description, amount) 😪

Reply:
Might be much better if this is assisted by a helper function?